src/v2/guide/reactivity.md

```diff
@@ -4,80 +4,80 @@ type: guide
 
-The getter/setters are invisible to the user, but under the hood they enable Vue to perform dependency-tracking and change-notification when properties are accessed or modified. One caveat is that browser consoles format getter/setters differently when converted data objects are logged, so you may want to install [vue-devtools](https://github.com/vuejs/vue-devtools) for a more inspection-friendly interface.
+Các getter và setter có thể được nhìn thấy bởi người dùng, nhưng sâu hơn nữa, nó cho phép Vue thực hiện theo dõi các phụ thuộc và thay đổi thông báo khi các thuộc tính được truy cập hoặc sửa đổi. Một trong những hạn chế của việc này là trình duyệt set in ra định dạng getter / setters khác nhau khi chuyển đổi các đối tượng, do đó khuyến khích các bạn cài đặt [vue-devtools](https://github.com/vuejs/vue-devtools) để có một cái nhìn sâu hơn về ứng dụng của mình.
 
 Every component instance has a corresponding **watcher** instance, which records any properties "touched" during the component's render as dependencies. Later on when a dependency's setter is triggered, it notifies the watcher, which in turn causes the component to re-render.
 
 ![Reactivity Cycle](/images/data.png)
 
-## Change Detection Caveats
+## Hạn chế của việc phát hiện thay đổi
 
-Due to the limitations of modern JavaScript (and the abandonment of `Object.observe`), Vue **cannot detect property addition or deletion**. Since Vue performs the getter/setter conversion process during instance initialization, a property must be present in the `data` object in order for Vue to convert it and make it reactive. For example:
+Do những hạn chế của JavaScript hiện tại (và việc `Object.observe` không còn được sử dụng tới), Vue **không thể phát hiện sự thêm hoặc xoá của một thuộc tính nào đó**. Vì Vue thực hiện quá trình chuyển đổi getter / setter trong quá trình khởi tạo thực thể, một thuộc tính bất kì phải được khai báo trong đối tượng `data` để Vue có thể thực hiện chuyển đổi và làm nó trở nên `reactive`. Ví dụ:
 
 ``` js
 var vm = new Vue({
   data: {
     a: 1
   }
 })
-// `vm.a` is now reactive
+// `vm.a` giờ đã trở thành reactive
 
 vm.b = 2
-// `vm.b` is NOT reactive
+// `vm.b` không phải `reactive`
 ```
 
-Vue does not allow dynamically adding new root-level reactive properties to an already created instance. However, it's possible to add reactive properties to a nested object using the `Vue.set(object, key, value)` method:
+Vue không cho phép thêm động các thuộc tính `reactive` ở mức `root` vào một thực thể đã được tạo từ trước. Tuy nhiên, có thể thêm các thuộc tính `reactive` vào một đối tượng lồng nhau bằng cách sử dụng `Vue.set(object, key, value)`:
 
 ``` js
 Vue.set(vm.someObject, 'b', 2)
 ```
 
-You can also use the `vm.$set` instance method, which is an alias to the global `Vue.set`:
+Bạn cũng có thể sử dụng `vm.$set`, đây là một bí danh (alias) toàn cầu của `Vue.set`:
```

Comment:
"global" trong tin học thường được dịch là "toàn cục" (đối với "local" là "cục bộ") chứ không phải "toàn cầu."